### PR TITLE
docs: add apisix-gitops-adc.md

### PR DESCRIPTION
**Pull request by Yilialinn**
Fixes: #[Add issue number here]

Changes:

Screenshots of the change:

### CODE REVIEW

File: blog/zh/blog/2023/10/07/apisix-gitops-adc.md [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+keywords:
+  - 开源社区
+  - API 网关
+  - Apache APISIX
+description: APISIX 引入了新的周边工具：支持 GitOps 声明式，帮助用户在非 Kubernetes 环境中以声明式的方式进行各种集成。
+tags: [Community]
+image: https://static.apiseven.com/2022/10/19/634f6677742a1.png
+---
+
+随着云原生和微服务的普及，API 网关成为了连接和管理不同微服务之间的重要组件。然而，随着服务数量的增加和变更的频繁发生，传统的命令式配置方式已经变得难以管理和维护。GitOps 是一种基于版本控制系统和自动化工作流程的运维模式，它将配置和策略以声明式的方式进行管理，从而提供了更高效、可靠和可追溯的操作方式。
```
Comment (SylviaBABY) [inline blog/zh/blog/2023/10/07/apisix-gitops-adc.md]:
```suggestion
随着云原生和微服务的普及，API 网关成为了连接和管理不同微服务之间的重要组件。然而，随着服务数量的增加和频繁变更的场景需求，传统的命令式配置方式已经变得难以管理和维护。GitOps 是一种基于版本控制系统和自动化工作流程的运维模式，它将配置和策略以声明式的方式进行管理，从而提供了更高效、可靠和可追溯的操作方式。
```

File: blog/zh/blog/2023/10/07/apisix-gitops-adc.md [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+
+为了提高开发者的效率和操作的可靠性，APISIX 引入了新的周边工具：支持 GitOps 声明式。APISIX 通过支持 GitOps 声明式特性进一步加强了与现代化开发和运维流程的整合。
+
+## 为什么支持 GitOps 声明式配置
+
+尽管 APISIX 提供了 Stand-alone 模式，可以用 YAML 文件为 APISIX 提供配置，但 APISIX 缺乏与 CICD 等相关生态系统，如 Jenkins、ArgoCD 等生态的完整集成。APISIX Ingress Controller 项目在这方面进行了许多对接和集成工作，但 APISIX 本身在裸金属或虚拟机等非 Kubernetes 环境中使用时，并没有提供一套完整的声明式工具来支持 GitOps。
```
Comment (SylviaBABY) [inline blog/zh/blog/2023/10/07/apisix-gitops-adc.md]:
```suggestion
尽管 APISIX 提供了 Stand-alone 模式，可以用 YAML 文件为 APISIX 提供配置，但 APISIX 缺乏与 CI/CD 等相关生态系统，如 Jenkins、ArgoCD 等生态的完整集成。APISIX Ingress Controller 项目在这方面进行了许多对接和集成工作，但 APISIX 本身在裸金属或虚拟机等非 Kubernetes 环境中使用时，并没有提供一套完整的声明式工具来支持 GitOps。
```

File: blog/zh/blog/2023/10/07/apisix-gitops-adc.md [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+
+3. 简化多环境管理：在开发、测试和生产环境中，往往需要管理不同的配置。通过使用 GitOps，可以轻松创建不同的分支或标签来管理不同环境的配置文件，从而实现环境之间的一致性，并减少了手动配置的错误。
+
+4. 促进团队协作：GitOps 的工作流程可以促进团队成员之间的协作和交流。开发者可以通过提交合并请求、代码审查等方式共同参与到 API 网关配置的开发和维护中，提高团队的协作效率和代码质量。
+
+## APISIX 如何支持 GitOps 声明式配置
+
+### ADC 及其使用场景和功能
+
+ADC 全称是 APISIX declarative CLI ， 是一个 APISIX 的声明式配置工具。APISIX 通过 ADC 帮助用户在非 Kubernetes 环境中以声明式的方式进行各种集成。
```
Comment (SylviaBABY) [inline blog/zh/blog/2023/10/07/apisix-gitops-adc.md]:
```suggestion
ADC 全称是 APISIX declarative CLI ，是一个 APISIX 的声明式配置工具。APISIX 通过 ADC 帮助用户在非 Kubernetes 环境中以声明式的方式进行各种集成。
```

File: blog/zh/blog/2023/10/07/apisix-gitops-adc.md [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+- OpenAPI 转换：ADC 可以将 OpenAPI 规范文件转换成 APISIX 的配置文件。
+- 运行诊断：ADC 支持 ping 等诊断命令，帮助调试 ADC 与网关的连接。
+
+ADC 本质上提供了一套声明式的 APISIX 配置和管理能力，不需要手动调用 admin API 或者通过 Dashboard 等，通过简单的命令就能实现配置的同步。
+
+### APISIX 如何使用 ADC 实现声明式配置
+
+#### 安装 APISIX 和 ADC
+
+参考 APISIX 文档，安装好 APISIX。再使用 go install 命令将 ADC 二进制安装到 $GOPATH/bin 目录：
```
Comment (SylviaBABY) [inline blog/zh/blog/2023/10/07/apisix-gitops-adc.md]:
```suggestion
参考 APISIX 文档，安装好 APISIX。再使用 `go install` 命令将 ADC 二进制安装到 $GOPATH/bin 目录：
```

File: blog/zh/blog/2023/10/07/apisix-gitops-adc.md [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+
+然后，您可以将输出复制粘贴到你的 `.zshrc` 文件中，之后在你使用 `adc` 时，它将显示提示。
+
+ADC 仍处于初级阶段，并且正在不断改进。要了解有关该项目的更多信息、报告错误或建议功能，请访问 [github.com/api7/adc](github.com/api7/adc)。
+
+## 总结
+
+APISIX 通过使用声明式配置工具 ADC 支持 GitOps 声明式特性，提供了一种更简化、可靠和可追溯的管理方式，使开发者能够更高效地管理和部署 API 网关的配置。这项新特性为团队协作、环境一致性和配置管理带来了许多好处，为构建可靠的云原生架构提供了有力支持。
+
+用户能在非 Kubernetes 环境中与 Jenkins、ArgoCD 等工具进行良好的对接，并使用 GitOps 的内部 CICD 方式来控制 APISIX 的各种行为，实现多环境发布等功能。
```
Comment (SylviaBABY) [inline blog/zh/blog/2023/10/07/apisix-gitops-adc.md]:
```suggestion
用户能在非 Kubernetes 环境中与 Jenkins、ArgoCD 等工具进行良好的对接，并使用 GitOps 的内部 CI/CD 方式来控制 APISIX 的各种行为，实现多环境发布等功能。
```

File: blog/zh/blog/2023/10/07/apisix-gitops-adc.md [hunk 6]
```diff
@@ -0,0 +1,322 @@
...
+keywords:
+  - 开源社区
+  - API 网关
+  - Apache APISIX
+description: APISIX 引入了新的周边工具：支持 GitOps 声明式，帮助用户在非 Kubernetes 环境中以声明式的方式进行各种集成。
+tags: [Community]
+image: https://static.apiseven.com/2022/10/19/634f6677742a1.png
+---
+
+APISIX 引入了新的周边工具：支持 GitOps 声明式，帮助用户在非 Kubernetes 环境中以声明式的方式进行各种集成。
```
Comment (saltism) [inline blog/zh/blog/2023/10/07/apisix-gitops-adc.md]: 我感觉这句话不是特别通。“工具”可能是一个客观存在的插件、软件，“支持 GitOps 声明式“只能算是功能或特性。

File: blog/zh/blog/2023/10/07/apisix-gitops-adc.md [hunk 7]
```diff
@@ -0,0 +1,322 @@
...
+
+尽管 APISIX 提供了 stand-alone 模式，可以用 YAML 文件为 APISIX 提供配置，但 APISIX 缺乏与 CI/CD 等相关生态系统，如 Jenkins、ArgoCD 等生态的完整集成。APISIX Ingress Controller 项目在这方面进行了许多对接和集成工作，但 APISIX 本身在裸金属或虚拟机等非 Kubernetes 环境中使用时，并没有提供一套完整的声明式工具来支持 GitOps。
+
+在传统的 API 网关管理中，配置和策略通常以命令式的方式进行操作，需要通过命令行工具或管理界面手动修改。传统的管理方式方式存在一些挑战，例如：
+
+- **配置管理繁琐**：手动修改配置容易出错，尤其当网关规模庞大时。
+- **可追溯性差**：难以追踪配置的变更历史和版本控制。
+- **缺乏一致性**：多个环境之间的配置差异导致开发、测试和生产环境的不一致性。
+
+APISIX 开发团队意识到支持 GitOps 声明式有以下几点好处，能有效应对这些挑战：
```
Comment (saltism) [inline blog/zh/blog/2023/10/07/apisix-gitops-adc.md]:
```suggestion
APISIX 开发团队意识到，支持 GitOps 声明式有以下几点好处：
```

File: blog/zh/blog/2023/10/07/apisix-gitops-adc.md [hunk 8]
```diff
@@ -0,0 +1,322 @@
...
+为了提高开发者的效率和操作的可靠性，APISIX 引入了新的周边工具：支持 GitOps 声明式。APISIX 通过支持 GitOps 声明式特性进一步加强了与现代化开发和运维流程的整合。
+
+## 为什么支持 GitOps 声明式配置
+
+尽管 APISIX 提供了 stand-alone 模式，可以用 YAML 文件为 APISIX 提供配置，但 APISIX 缺乏与 CI/CD 等相关生态系统，如 Jenkins、ArgoCD 等生态的完整集成。APISIX Ingress Controller 项目在这方面进行了许多对接和集成工作，但 APISIX 本身在裸金属或虚拟机等非 Kubernetes 环境中使用时，并没有提供一套完整的声明式工具来支持 GitOps。
+
+在传统的 API 网关管理中，配置和策略通常以命令式的方式进行操作，需要通过命令行工具或管理界面手动修改。传统的管理方式方式存在一些挑战，例如：
+
+- **配置管理繁琐**：手动修改配置容易出错，尤其当网关规模庞大时。
+- **可追溯性差**：难以追踪配置的变更历史和版本控制。
```
Comment (saltism) [inline blog/zh/blog/2023/10/07/apisix-gitops-adc.md:35]:
```suggestion
- **可追溯性差**：变更历史和版本控制难以追踪配置。
```

File: blog/zh/blog/2023/10/07/apisix-gitops-adc.md [hunk 9]
```diff
@@ -0,0 +1,322 @@
...
+APISIX 引入了新的周边工具：支持 GitOps 声明式，帮助用户在非 Kubernetes 环境中以声明式的方式进行各种集成。
+<!--truncate-->
+
+随着云原生和微服务的普及，API 网关成为了连接和管理不同微服务之间的重要组件。然而，随着服务数量的增加和场景需求的频繁变更，传统的命令式配置方式已经变得难以管理和维护。GitOps 是一种基于版本控制系统和自动化工作流程的运维模式，它将配置和策略以声明式的方式进行管理，从而提供了更高效、可靠和可追溯的操作方式。
+
+为了提高开发者的效率和操作的可靠性，APISIX 引入了新的周边工具：支持 GitOps 声明式。APISIX 通过支持 GitOps 声明式特性进一步加强了与现代化开发和运维流程的整合。
+
+## 为什么支持 GitOps 声明式配置
+
+尽管 APISIX 提供了 stand-alone 模式，可以用 YAML 文件为 APISIX 提供配置，但 APISIX 缺乏与 CI/CD 等相关生态系统，如 Jenkins、ArgoCD 等生态的完整集成。APISIX Ingress Controller 项目在这方面进行了许多对接和集成工作，但 APISIX 本身在裸金属或虚拟机等非 Kubernetes 环境中使用时，并没有提供一套完整的声明式工具来支持 GitOps。
```
Comment (saltism) [inline blog/zh/blog/2023/10/07/apisix-gitops-adc.md]:
```suggestion
尽管 APISIX 提供了 stand-alone 模式，可以用 YAML 文件为 APISIX 提供配置，但 APISIX 缺乏与 CI/CD 等相关生态系统，如 Jenkins、ArgoCD 等生态的完整集成。虽然 APISIX Ingress Controller 项目在这方面进行了许多对接和集成工作，但 APISIX 本身在裸金属或虚拟机等非 Kubernetes 环境中使用时，并没有提供一套完整的声明式工具来支持 GitOps。
```

File: blog/zh/blog/2023/10/07/apisix-gitops-adc.md [hunk 10]
```diff
@@ -0,0 +1,322 @@
...
+version: ""
+```
+
+正如我们所看到的，配置是不完整的，仍然需要手动添加大量配置。我们正在改进 ADC，以弥补 OpenAPI 定义和可以直接映射到 APISIX 配置之间的差距。
+
+### 提示：使用自动完成
+
+ADC 可以为你提供很多帮助，而且功能列表一定会增加。要了解如何使用任何子命令，可以使用 `--help` 或 `-h` 标志，它将显示该子命令的文档。
+
+为了让它变得更简单，您可以使用 `completion` 子命令为你的 `shell` 环境生成一个自动完成脚本。例如，如果使用的是 zsh shell，则可以运行：
```
Comment (saltism) [inline blog/zh/blog/2023/10/07/apisix-gitops-adc.md]:
```suggestion
为了让它变得更简单，您可以使用 `completion` 子命令为您的 `shell` 环境生成一个自动完成脚本。例如，如果使用的是 zsh shell，则可以运行：
```

File: blog/zh/blog/2023/10/07/apisix-gitops-adc.md [hunk 11]
```diff
@@ -0,0 +1,322 @@
...
+    name: ""
+    nodes: null
+version: ""
+```
+
+正如我们所看到的，配置是不完整的，仍然需要手动添加大量配置。我们正在改进 ADC，以弥补 OpenAPI 定义和可以直接映射到 APISIX 配置之间的差距。
+
+### 提示：使用自动完成
+
+ADC 可以为你提供很多帮助，而且功能列表一定会增加。要了解如何使用任何子命令，可以使用 `--help` 或 `-h` 标志，它将显示该子命令的文档。
```
Comment (saltism) [inline blog/zh/blog/2023/10/07/apisix-gitops-adc.md:306]:
```suggestion
ADC 可以为您提供很多帮助，而且功能列表一定会增加。要了解如何使用任何子命令，可以使用 `--help` 或 `-h` 标志，它将显示该子命令的文档。
```

File: blog/zh/blog/2023/10/07/apisix-gitops-adc.md [hunk 12]
```diff
@@ -0,0 +1,322 @@
...
+      summary: Returns the IP address of the requester
+      operationId: getIP
+      responses:
+        "200":
+          description: Successfully return IP
+          content:
+            application/json: {}
+```
+
+你可以使用子命令 `openapi2apisix` 将其转换为 APISIX 配置，如下所示：
```
Comment (saltism) [inline blog/zh/blog/2023/10/07/apisix-gitops-adc.md:266]:
```suggestion
您可以使用子命令 `openapi2apisix` 将其转换为 APISIX 配置，如下所示：
```

File: blog/zh/blog/2023/10/07/apisix-gitops-adc.md [hunk 13]
```diff
@@ -0,0 +1,322 @@
...
+adc diff -f config.yaml
+```
+
+在应用配置之前，能够看到添加和删除的配置，也能了解到更改的内容。
+
+### 将 OpenAPI 定义转换为 APISIX 配置
+
+ADC 还支持使用 OpenAPI 定义。ADC 允许将 [OpenAPI 格式](https://spec.openapis.org/oas/v3.0.0)的定义转换为 APISIX 配置。
+
+例如，如果你以 OpenAPI 格式记录了 API，如下所示：
```
Comment (saltism) [inline blog/zh/blog/2023/10/07/apisix-gitops-adc.md:225]:
```suggestion
例如，如果您以 OpenAPI 格式记录了 API，如下所示：
```

File: blog/zh/blog/2023/10/07/apisix-gitops-adc.md [hunk 14]
```diff
@@ -0,0 +1,322 @@
...
+
+你可以在[发布页](https://github.com/api7/adc/releases/tag/v0.2.0)上找到其他操作系统的二进制文件。将来，这些文件将发布在像 Homebrew 这样的软件包管理工具上。
+
+运行下列代码，确认 `adc` 已经安装好：
+
+```shell
+adc --help
+```
+
+如果一切正常，你将看到可用子命令的列表以及如何使用它们。
```
Comment (saltism) [inline blog/zh/blog/2023/10/07/apisix-gitops-adc.md]:
```suggestion
如果一切正常，您将看到可用子命令的列表以及如何使用它们的方法。
```

File: blog/zh/blog/2023/10/07/apisix-gitops-adc.md [hunk 15]
```diff
@@ -0,0 +1,322 @@
...
+
+如果你没有安装 Go，则可以下载最新版本的 `adc` 二进制来运行，把它加到你的 `/bin` 文件夹。
+
+```shell
+wget https://github.com/api7/adc/releases/download/v0.2.0/adc_0.2.0_linux_amd64.tar.gz
+tar -zxvf adc_0.2.0_linux_amd64.tar.gz
+mv adc /usr/local/bin/adc
+```
+
+你可以在[发布页](https://github.com/api7/adc/releases/tag/v0.2.0)上找到其他操作系统的二进制文件。将来，这些文件将发布在像 Homebrew 这样的软件包管理工具上。
```
Comment (saltism) [inline blog/zh/blog/2023/10/07/apisix-gitops-adc.md:90]:
```suggestion
您可以在[发布页](https://github.com/api7/adc/releases/tag/v0.2.0)上找到其他操作系统的二进制文件。将来，这些文件将发布在像 Homebrew 这样的软件包管理工具上。
```

File: blog/zh/blog/2023/10/07/apisix-gitops-adc.md [hunk 16]
```diff
@@ -0,0 +1,322 @@
...
+
+### 安装 APISIX 和 ADC
+
+参考 [APISIX 文档](https://apisix.apache.org/zh/docs/apisix/getting-started/README/)，安装好 APISIX。再使用 `go install` 命令将 ADC 二进制安装到 `$GOPATH/bin` 目录：
+
+```shell
+go install github.com/api7/adc@latest
+```
+
+增加这行代码到你的 `$PATH` 环境变量：
```
Comment (saltism) [inline blog/zh/blog/2023/10/07/apisix-gitops-adc.md:76]:
```suggestion
增加这行代码到您的 `$PATH` 环境变量：
```

File: blog/zh/blog/2023/10/07/apisix-gitops-adc.md [hunk 17]
```diff
@@ -0,0 +1,322 @@
...
+
+ADC 可以为你提供很多帮助，而且功能列表一定会增加。要了解如何使用任何子命令，可以使用 `--help` 或 `-h` 标志，它将显示该子命令的文档。
+
+为了让它变得更简单，您可以使用 `completion` 子命令为你的 `shell` 环境生成一个自动完成脚本。例如，如果使用的是 zsh shell，则可以运行：
+
+```shell
+adc completion zsh
+```
+
+然后，您可以将输出复制粘贴到你的 `.zshrc` 文件中，之后在你使用 `adc` 时，它将显示提示。
```
Comment (saltism) [inline blog/zh/blog/2023/10/07/apisix-gitops-adc.md]:
```suggestion
然后，您可以将输出复制粘贴到你的 `.zshrc` 文件中，之后在您使用 `adc` 时，它将显示提示。
```

File: blog/en/blog/2023/10/08/apisix-gitops-adc.md [hunk 18]
```diff
@@ -0,0 +1,326 @@
...
+To enhance developing efficiency and operational reliability, APISIX has introduced a new tool that supports GitOps in a declarative manner. By embracing the declarative nature of GitOps, APISIX strengthens its integration with modern development and operational workflows. This integration enables smoother collaboration between developers and operations teams, promoting efficient and reliable management of the APISIX platform.
+
+## Why Does APISIX Support GitOps Declarative Configuration
+
+Although APISIX offers a stand-alone mode that allows configuration through YAML files, it lacks seamless integration with related ecosystems such as CI/CD tools like Jenkins and ArgoCD. While the APISIX Ingress Controller project has made significant strides in this area, APISIX itself does not provide a comprehensive set of declarative tools to support GitOps when used in non-Kubernetes environments such as bare metal or virtual machines.
+
+In traditional API gateway management, configurations and policies are typically manipulated using imperative methods, requiring manual modifications through command-line tools or management interfaces. This approach poses several challenges, including:
```
Comment (saltism) [inline blog/en/blog/2023/10/08/apisix-gitops-adc.md]:
```suggestion
In traditional API gateway management, configurations and policies are typically manipulated using imperative methods, requiring manual modifications through command-line tools or management interfaces. This approach poses several challenges:
```

File: blog/zh/blog/2023/10/07/apisix-gitops-adc.md [hunk 19]
```diff
@@ -0,0 +1,322 @@
...
+
+- **配置连接信息**：ADC 可以配置要连接的 APISIX 实例的地址、端口、登录令牌等信息。
+- **验证配置**：ADC 提供了验证 APISIX 配置文件语法的功能。
+- **配置同步**：可以通过 ADC 将本地的配置文件同步到 APISIX 实例。
+- **配置导出**：可以通过 ADC 从 APISIX 实例导出配置文件。
+- **比较配置差异**：ADC 可以比较本地配置和 APISIX 实例上的配置差异。
+- **OpenAPI 转换**：ADC 可以将 OpenAPI 规范文件转换成 APISIX 的配置文件。
+- **运行诊断**：ADC 支持 ping 等诊断命令，帮助调试 ADC 与网关的连接。
+
+ADC 本质上提供了一套声明式的 APISIX 配置和管理能力，不需要手动调用 admin API 或者通过 Dashboard 等，通过简单的命令就能实现配置的同步。
```
Comment (saltism) [inline blog/zh/blog/2023/10/07/apisix-gitops-adc.md]:
```suggestion
ADC 本质上提供了一套声明式的 APISIX 配置和管理能力，通过简单的命令就能实现配置的同步，不需要手动调用 admin API 或者通过 Dashboard 等。

```

File: blog/en/blog/2023/10/08/apisix-gitops-adc.md [hunk 20]
```diff
@@ -0,0 +1,326 @@
...
+
+To enhance developing efficiency and operational reliability, APISIX has introduced a new tool that supports GitOps in a declarative manner. By embracing the declarative nature of GitOps, APISIX strengthens its integration with modern development and operational workflows. This integration enables smoother collaboration between developers and operations teams, promoting efficient and reliable management of the APISIX platform.
+
+## Why Does APISIX Support GitOps Declarative Configuration
+
+Although APISIX offers a stand-alone mode that allows configuration through YAML files, it lacks seamless integration with related ecosystems such as CI/CD tools like Jenkins and ArgoCD. While the APISIX Ingress Controller project has made significant strides in this area, APISIX itself does not provide a comprehensive set of declarative tools to support GitOps when used in non-Kubernetes environments such as bare metal or virtual machines.
```
Comment (saltism) [inline blog/en/blog/2023/10/08/apisix-gitops-adc.md]: Tense?

File: blog/en/blog/2023/10/08/apisix-gitops-adc.md [hunk 21]
```diff
@@ -0,0 +1,326 @@
...
+
+In traditional API gateway management, configurations and policies are typically manipulated using imperative methods, requiring manual modifications through command-line tools or management interfaces. This approach poses several challenges, including:
+
+- **Cumbersome Configuration Management**: Manual modifications of configurations are prone to errors, especially when dealing with large-scale gateways.
+
+- **Poor Traceability**: Tracking the change history and version control of configurations becomes difficult.
+
+- **Lack of Consistency**: Configuration discrepancies among multiple environments result in inconsistencies between development, testing, and production environments.
+
+The APISIX development team recognizes several benefits of supporting GitOps in a declarative manner to effectively address these challenges:
```
Comment (saltism) [inline blog/en/blog/2023/10/08/apisix-gitops-adc.md]: Also tense. Would it be better to use "recognized"?

File: blog/en/blog/2023/10/08/apisix-gitops-adc.md [hunk 22]
```diff
@@ -0,0 +1,326 @@
...
+
+1. **Improved Developer Efficiency**: By using GitOps with a declarative configuration approach, developers can directly manage API gateway configurations by modifying and committing configuration files in the code repository. This method aligns with the development workflow that developers are familiar with, reducing the learning curve and tool-switching costs, and thus enhancing developer productivity.
+
+2. **Enhanced Operational Reliability**: Storing configuration files in a version control system ensures consistency and reliability. Each configuration change can be traced through its change history, and it becomes easy to roll back to a previous configuration state. This enables traceability, and auditability, and reduces the risk of failures caused by human errors.
```
Comment (saltism) [inline blog/en/blog/2023/10/08/apisix-gitops-adc.md]:
```suggestion
2. **Enhanced Operational Reliability**: Storing configuration files in a version control system ensures consistency and reliability. Each configuration change can be traced through its change history, and it becomes easy to roll back to a previous configuration state. This function enables traceability and auditability, and reduces the risk of failures caused by human errors.
```

File: blog/en/blog/2023/10/08/apisix-gitops-adc.md [hunk 23]
```diff
@@ -0,0 +1,326 @@
...
+
+You can find binaries for other operating systems on the [releases page](https://github.com/api7/adc/releases/tag/v0.2.0). In the future, these files will be published on package management tools like Homebrew.
+
+Run the following code to confirm that `adc` has been installed:
+
+```shell
+adc --help
+```
+
+If everything goes well, you will see a list of available subcommands and how to use them.
```
Comment (saltism) [inline blog/en/blog/2023/10/08/apisix-gitops-adc.md]: 这里“How to use them“是一个会pop up的section吗？如果不是，感觉还是改成”usage" or "using guide"之类的词会好一些。

File: blog/en/blog/2023/10/08/apisix-gitops-adc.md [hunk 24]
```diff
@@ -0,0 +1,326 @@
...
+          weight: 1
+routes:
+  - name: httpbin-route
+    service_id: httpbin-service
+    uri: "/anything"
+    methods:
+      - GET
+```
+
+Once the ADC is connected to the running APISIX instance, we can use it to validate this configuration before applying it by running:
```
Comment (saltism) [inline blog/en/blog/2023/10/08/apisix-gitops-adc.md]: 感觉操作指南也是"we""you"混用着来，统一一下会不会更好呢？

File: blog/zh/blog/2023/10/07/apisix-gitops-adc.md [hunk 25]
```diff
@@ -0,0 +1,322 @@
...
+      - GET
+```
+
+在将此配置与 APISIX 同步之前，ADC 允许您检查它与现有 APISIX 配置之间的差异。可以运行以下操作：
+
+```shell
+adc diff -f config.yaml
+```
+
+在应用配置之前，能够看到添加和删除的配置，也能了解到更改的内容。
```
Comment (saltism) [inline blog/zh/blog/2023/10/07/apisix-gitops-adc.md:219]:
```suggestion
在应用配置之前，您能够看到添加和删除的配置，也能确认已更改的内容。
```

File: blog/en/blog/2023/10/08/apisix-gitops-adc.md [hunk 26]
```diff
@@ -0,0 +1,326 @@
...
+      - GET
+```
+
+Before synchronizing this configuration with APISIX, ADC allows you to check the differences between it and the existing APISIX configuration. You can run the following operations:
+
+```shell
+adc diff -f config.yaml
+```
+
+You can see the added and deleted configurations and understand the changes before applying the configuration.
```
Comment (saltism) [inline blog/en/blog/2023/10/08/apisix-gitops-adc.md]:
```suggestion
You can see the added and deleted configurations and check the changes before applying the configuration.
```

File: blog/en/blog/2023/10/08/apisix-gitops-adc.md [hunk 27]
```diff
@@ -0,0 +1,326 @@
...
+
+
+### Converting OpenAPI Definitions to APISIX Configurations
+
+ADC also supports the use of [OpenAPI definitions](https://spec.openapis.org/oas/v3.0.0). ADC allows the conversion of OpenAPI format definitions into APISIX configurations.
+
+For example, if you documented your API in OpenAPI format like this:
```
Comment (saltism) [inline blog/en/blog/2023/10/08/apisix-gitops-adc.md]:
```suggestion
For example, if you document your API in OpenAPI format like this:
```

File: blog/en/blog/2023/10/08/apisix-gitops-adc.md [hunk 28]
```diff
@@ -0,0 +1,326 @@
...
+
+You can then copy and paste the output into your `.zshrc` file and it will start showing hints when you use `adc`.
+
+ADC is still in its infancy and is constantly being improved. To learn more about the project, report a bug, or suggest a feature, visit [github.com/api7/adc](github.com/api7/adc).
+
+## Summary
+
+By using the declarative configuration tool ADC, APISIX provides a more simplified, reliable, and traceable management method, allowing developers to manage and deploy API gateway configurations more efficiently. This new feature brings many benefits to team collaboration, environmental consistency, and configuration management, providing strong support for building reliable cloud-native architectures.
+
+Users can connect well with tools such as Jenkins and ArgoCD in non-Kubernetes environments, and use GitOps's internal CI/CD method to control various behaviors of APISIX to achieve functions such as multi-environment release.
```
Comment (saltism) [inline blog/en/blog/2023/10/08/apisix-gitops-adc.md]:
```suggestion
In non-Kubernetes environments, users can seamlessly integrate tools like Jenkins and ArgoCD. They can leverage GitOps' internal CI/CD approach to manage various aspects of APISIX, enabling functions like multi-environment releases.
```

**Review by saltism — CHANGES_REQUESTED** [review]

File: blog/zh/blog/2023/10/07/apisix-gitops-adc.md (hunk 14 above)
Comment (Yilialinn) [reply blog/zh/blog/2023/10/07/apisix-gitops-adc.md:98]: 这样改是病句了哦，这里我改为“使用它们的方法”吧

Comment (Yilialinn) [reply blog/zh/blog/2023/10/07/apisix-gitops-adc.md]: Wrong expression in Chinese. I'll change it into "使用它们的方法".

File: blog/zh/blog/2023/10/07/apisix-gitops-adc.md [hunk 29]
```diff
@@ -0,0 +1,323 @@
...
+tags: [Community]
+image: https://static.apiseven.com/2022/10/19/634f6677742a1.png
+---
+
+APISIX 引入了新的周边工具：支持 GitOps 声明式，帮助用户在非 Kubernetes 环境中以声明式的方式进行各种集成。
+<!--truncate-->
+
+随着云原生和微服务的普及，API 网关成为了连接和管理不同微服务之间的重要组件。然而，随着服务数量的增加和场景需求的频繁变更，传统的命令式配置方式已经变得难以管理和维护。GitOps 是一种基于版本控制系统和自动化工作流程的运维模式，它将配置和策略以声明式的方式进行管理，从而提供了更高效、可靠和可追溯的操作方式。
+
+为了提高开发者的效率和操作的可靠性，APISIX 引入了新的周边工具：支持 GitOps 声明式。APISIX 通过支持 GitOps 声明式特性进一步加强了与现代化开发和运维流程的整合。
```
Comment (yzeng25) [inline blog/zh/blog/2023/10/07/apisix-gitops-adc.md]: ditto https://github.com/apache/apisix-website/pull/1703/files#r1349648273

File: blog/zh/blog/2023/10/07/apisix-gitops-adc.md (hunk 16 above)
Comment (yzeng25) [reply blog/zh/blog/2023/10/07/apisix-gitops-adc.md:76]: I think '你' is fine unless you'll change the remaining '你's to '您's

File: blog/zh/blog/2023/10/07/apisix-gitops-adc.md [hunk 30]
```diff
@@ -0,0 +1,323 @@
+---
```
Comment (yzeng25) [inline blog/zh/blog/2023/10/07/apisix-gitops-adc.md:1]: You should keep both file path with the same pattern, i.e. both should be `./2023/10/08/.` or `./2023/10/07/.`
截屏2023-10-08 20 12 04

**Comment by SylviaBABY** [comment]
pls fix the Lint errors
image

File: blog/en/blog/2023/10/07/apisix-gitops-adc.md [hunk 31]
```diff
@@ -0,0 +1,324 @@
...
+    title: Author
+    url: https://github.com/tao12345666333
+    image_url: https://avatars.githubusercontent.com/u/3264292?v=4
+  - name: Yilia Lin
+    title: Technical Writer
+    url: https://github.com/Yilialinn
+    image_url: https://avatars.githubusercontent.com/u/114121331?v=4
+keywords:
+  - 开源社区
+  - API 网关
```
Comment (yzeng25) [inline blog/en/blog/2023/10/07/apisix-gitops-adc.md]: Should these two keywords be in English?

**Review by yzeng25 — APPROVED** [review]

**Review by SylviaBABY — APPROVED** [review]

**Comment by yzeng25** [comment]
@SylviaBABY when merging pull request use 'squash and merge' to make it one commit next time.

**Comment by SylviaBABY** [comment]
> @SylviaBABY when merging pull request use 'squash and merge' to make it one commit next time.

sry to forget deleteing those infos……its too long🤣

**Pull request merged**